src/HMRC.ESFA.Levy.Api.Client/Services/PaymentStatusProcessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using HMRC.ESFA.Levy.Api.Types;
+
+namespace HMRC.ESFA.Levy.Api.Client.Services
+{
+    public class PaymentStatusProcessor : IPaymentStatusProcessor
+    {
+        private const int DayInMonthForSubmissionProcessing = 23;
+        private const int DayInMonthForSubmissionCutoff = 20;
+
+        public List<Declaration> ProcessDeclarationPaymentStatuses(List<Declaration> declarations, DateTime dateAdded)
+        {
+            var declarationsWithPaymentPeriod = declarations.Where(x => x.PayrollPeriod != null).ToList();
+
+            var distinctPeriodsInPayroll = declarationsWithPaymentPeriod
+                .Select(x => $@"{x.PayrollPeriod.Year}{x.PayrollPeriod.Month}")
+                .Distinct();
+
+            foreach (var period in distinctPeriodsInPayroll)
```

Comment:
this section is complicated, It's a hard to see what it's doing by just looking at it